lib/src/main/java/org/kpipe/processor/AvroMessageProcessor.java

```diff
@@ -0,0 +1,682 @@
+package org.kpipe.processor;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.lang.System.Logger;
+import java.lang.System.Logger.Level;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.avro.util.Utf8;
+
+/**
+ * Provides utility functions for processing Avro messages. This class contains common message
+ * processors for use with the MessageProcessorRegistry to transform and enhance Avro data in byte
+ * array format.
+ *
+ * <p>Example usage:
+ *
+ * <pre>{@code
+ * // Register a schema
+ * final var userSchemaJson = """
+ *   {
+ *     "type": "record",
+ *     "name": "User",
+ *     "fields": [
+ *       {"name": "name", "type": "string"},
+ *       {"name": "age", "type": "int"}
+ *     ]
+ *   }
+ *   """;
+ * AvroMessageProcessor.registerSchema("userSchema", userSchemaJson);
+ *
+ * // Create a pipeline using these processors
+ * Function<byte[], byte[]> pipeline = MessageProcessorRegistry.pipeline(
+ *     "parseAvro",
+ *     MessageProcessorRegistry.get("addTimestampAvro")
+ * );
+ *
+ * // Process a message
+ * byte[] result = pipeline.apply(avroBytes);
+ * }</pre>
+ */
+public class AvroMessageProcessor {
+
+  private AvroMessageProcessor() {}
+
+  private static final Logger LOGGER = System.getLogger(AvroMessageProcessor.class.getName());
+  private static final byte[] EMPTY_AVRO = new byte[0];
+  private static final ConcurrentHashMap<String, Schema> SCHEMA_REGISTRY = new ConcurrentHashMap<>();
```

Comment:
Wouldn't more typical be integration with external schema registry? i.e. confluent schema registry or Karapace etc..